runtime/src/accounts_cache.rs

```diff
@@ -64,7 +65,7 @@ impl SlotCacheInner {
         }
         let item = Arc::new(CachedAccountInner {
             account,
-            hash: RwLock::new(hash),
+            hash: RwLock::new(hash.map(|h| *h.borrow())),
```

Comment:
Only question for me here is how do I not copy here if the passed in type is a value and not &Hash?

Reply:
Hmmm I think for this you may have to use a `Cow`

Reply:
my [measurements](https://godbolt.org/z/qGjb83onM) here seem to indicate that Borrow<> does what I want and seems to generate the code I want for everything except it would be nice if I passed an owned Hash that that hash could be moved to CachedAccountInner. As it is, arrays of hashes are no longer being created unnecessarily. And, in this case, the code path that causes us to write to the cache should usually not have a hash value anyway, so it should be None, side stepping the problem completely.

Reply:
Note that as far as I can tell, the trait that I first implemented did allow no additional copies to be made, whereas using borrow causes a copy to be made in this specific line. We were previously making a copy up the call chain before calling the function in most callers anyway, so this is still far better.